src/main/scala/squid/ir/Lowering.scala

```diff
@@ -41,7 +41,29 @@ trait Lowering extends Transformer {
               val ruh = RuntimeUniverseHelpers
               val typ = r.typ.typeArgs.last // TODO be careful that a method implemented without parameter list can implement a method with an empty list (eg toString), which can make this crash
               base.rep(MethodApp(r, ruh.FunctionType.symbol(reps.size).toType member ruh.sru.TermName("apply") asMethod, Nil, Args(reps:_*)::Nil, typ))
-            case (r, args) => lastWords(s"Not supported yet: vararg $args") // TODO
+            case (r, ArgsVarargs(Args(reps @ _*), Args(vreps @ _*))) =>
+              val ruh = RuntimeUniverseHelpers
+              val functionArity = reps.size + vreps.size
+              val typ = r.typ.typeArgs.last
```

Comment:
Why would `typ` in the `Seq[typ]` type of the vararg argument correspond to the last type argument to the method!? Very wild assumption here 😛

Reply:
Hah, good catch, this is definitely the wrong type. I think I just copied it from the regular case above.

But now I'm confused: I saw in your PR that you changed the construction of the `typ` to use the least upper bound for the typ of the sequence constructor and that makes sense to me. 

But it's also used for the type of the final `methodApp` construction. Shouldn't that be a different type here? As far as I understood `r.typ.typeArgs.last` gives us the return type of the `Def`, right?

In that case I would change the code so that it uses the `lub` of the `vreps` for the `Seq` type and the `r.typ.typeArgs.last` as the last argument to `methodApp`.

Reply:
You're right, good catch! I made the modification too fast and forgot that `typ` was also used later on (in that humongous `MethodApp` line that I should I broken in two distinct lines anyway).

Also, relative to using `r.typ.typeArgs.last`: it took me some time to remember (I should have _definitely_ left a comment about such a non-trivial mechanism), but the actual reason such a type works as the method's return type is because at this point `r` has to be a function type, whose last type argument indicates its return type (so my own earlier "wild assumption" comment about that type was moot).

That's what happens when you come back to code you haven't touched in many months 😅

Reply:
Thanks, that makes sense. I added the comment to my commit ;)